SCClassLibrary/Common/Control/ServerStatus.sc

```diff
@@ -37,30 +37,75 @@ ServerStatusWatcher {
 	}
 
 	sendNotifyRequest { |flag = true|
-		var doneOSCFunc, failOSCFunc;
+		var doneOSCFunc, failOSCFunc, newClientID, newMaxLogins;
 		if(serverRunning.not) { ^this };
 		notified = flag;
-		if(server.userSpecifiedClientID.not) {
-			doneOSCFunc = OSCFunc({|msg|
-				if(flag) { server.clientID = msg[2] };
-				failOSCFunc.free;
-			}, '/done', server.addr, argTemplate:['/notify', nil]).oneShot;
-
-			failOSCFunc = OSCFunc({|msg|
-				doneOSCFunc.free;
+
+		// always request maxLogins and free clientID from scsynth
+		doneOSCFunc = OSCFunc({|msg|
```

Comment:
i would really recommend factoring this out into a separate method - perhaps you could do one each for `prConstructDoneOSCFunc()` and `prConstructFailOSCFunc()`.